src/Parsing/Impl/Ast/Comprehension.cs

```diff
@@ -45,6 +51,24 @@ public abstract class Comprehension : Expression {
                 res.Append(end);
             }
         }
+
+        #region IScopeNode
+
+        public virtual string Name => "<comprehension>";
+        public IScopeNode ParentNode { get; set; }
```

Comment:
`ParentScopeNode` since `ParentNode` sounds like node that is right above in the tree which is not the case for scopes.